app/models/concerns/taggable.rb

```diff
@@ -0,0 +1,17 @@
+module Taggable
```

Comment:
Yes, so the `cached_tags` is one table field defined for the `Group` and `Resource` models. The problem with `acts-as-taggable-on` is that the tags are saved after the record has been saved, therefore when we send the model for ES indexing, the tags are not available yet.

The only way I found to do this was like this:

1. New tag is added to record. Record is saved and goes for indexing (but nothing will be updated because no fields has changed, yet).
2. acts-as-taggable-on kicks in, saving the tag and linking it to the record.
3. Because `touch: true` is defined on the relation between the taggings and the records (that's the line you ask me to explain below), the tagging record will touch (updating the `updated_at` basically) the record.
4. The `after_touch` callback is triggered, the list of tags stored in the record is updated by querying for the tag list from the `Tags`/`Taggings` tables.
5. The `after_commit on [:update]` is finally triggered, replaces the `cached_tags` by `tags` in the `as_indexed_json` (because that's how we want it to be called in the ES index) and send it for ES re-indexing/update.

Oh and the `as_indexed_json` is the hash that will be turn into a JSON and sent to ES for indexing, here I just override it to replace the `cached_tags` key by `tags`. We use it in the `SearchIndex` class as follow:

```
def changes
    record.__elasticsearch__.as_indexed_json.select do |k, _|
      changed.include?(k.to_s)
    end
  end
```

Let me know if anything is unclear ;)